src/components/Api.js

```diff
@@ -0,0 +1,89 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+      this._baseUrl = baseUrl;
+      this._headers = headers;
+    }
+  
+    _checkServerResponse(res) {
```

Comment:
it's great to have a method for checking the server response, so the logic is not duplicated